google/generativeai/string_utils.py

```diff
@@ -14,9 +14,42 @@
 # limitations under the License.
 from __future__ import annotations
 
+import pprint
+import textwrap
+import dataclasses
+
 
 def strip_oneof(docstring):
     lines = docstring.splitlines()
     lines = [line for line in lines if ".. _oneof:" not in line]
     lines = [line for line in lines if "This field is a member of `oneof`_" not in line]
     return "\n".join(lines)
+
+
+def prettyprint(cls):
+    cls.__str__ = _prettyprint
+    cls.__repr__ = _prettyprint
+    return cls
+
+
+def _prettyprint(self):
+    """You can't use `__str__ = pprint.pformat`. That causes a recursion error.
```

Comment:
> reprlib.recursive_repr.

Nice

> [etils dataclasses](https://github.com/google/etils/tree/main/etils/edc#other-augmentations)

This would resolve 50% of the problem (splitting lines).
But the other half of the reason is to contract long lists/dicts/objects (TunedModels's snapshots).